controllers/imagejob/imagejob_controller.go

```diff
@@ -82,17 +95,67 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
 func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
-
-	// your logic here
 
-	// create eraser pod in every node like broadcast job does
-	/// get list of nodes and schdules the pods on each of the nodes using pod.spec.nodename
-	// eventually check if pod fits before doing that
-
-	//var imageJob eraserv1alpha1.ImageJob
-	//if err := r.Get
+	nodes := &v1.NodeList{}
+	err := r.List(context.TODO(), nodes)
+	if err != nil {
+		panic(err)
+	}
 
+	count := 0
+
+	for _, n := range nodes.Items {
```

Comment:
can we go through all the nodes and save it into a list instead of doing everything in this loop

Reply:
I think we would need access to the node name and its runtime, should we save those in a map then and iterate the map again to start the pods on each node?

Reply:
we talked about saving this to a map for memory optimization